compass/landice/tests/slm/circ_icesheet/setup_mesh.py

```diff
@@ -0,0 +1,409 @@
+import os
+import shutil
+
+import mpas_tools.io
+import netCDF4
+import numpy as np
+from mpas_tools.io import write_netcdf
+from mpas_tools.logging import check_call
+from mpas_tools.mesh.conversion import cull
+from mpas_tools.planar_hex import make_planar_hex_mesh
+from mpas_tools.scrip.from_mpas import scrip_from_mpas
+from mpas_tools.translate import center, translate
+from netCDF4 import Dataset as NetCDFFile
+
+from compass.model import make_graph_file
+from compass.step import Step
+
+
+class SetupMesh(Step):
+    """
+    A step for creating a mesh with initial conditions plus forcing file for
+    circular icesheet test
+
+    Attributes
+    ----------
+    res : str
+        Resolution of MALI mesh
+
+    nglv : str
+        Number of Gauss-Legendre nodes in latitude in the SLM
+    """
+
+    def __init__(self, test_case, name, res, nglv):
+        """
+        Create the step
+
+        Parameters
+        ----------
+        test_case : compass.TestCase
+            The test case this step belongs to
+
+        res : str
+            Resolution of MALI mesh
+
+        nglv : str
+            Number of Gauss-Legendre nodes in latitude in the SLM
+        """
+        super().__init__(test_case=test_case, name=name)
+
+        self.add_output_file(filename='graph.info')
+        self.add_output_file(filename='landice_grid.nc')
+
+        self.res = res
+        self.nglv = nglv
+    # no setup() method is needed
+
+    def run(self):
+        """
+        Run this step of the test case
+        """
+        logger = self.logger
+        config = self.config
+        section = config['circ_icesheet']
+
+        lx = section.getfloat('lx')
+        ly = section.getfloat('ly')
+        dc = float(self.res) * 1000.0
```

Comment:
In the cfg file, lx and ly are defined in m, but it looks like here there is an assumption they are km.

Reply:
That's correct. I set it that way because I wanted `mali_res` to be integer type so it can be read and used in creating a step in format of `{mali_res}_slm{nglv}`. I guess it would have been cleaner to use 'meter' convention everywhere, and then change the data type from float to integer in `__init__.py`...